core/src/main/java/org/elasticsearch/threadpool/ThreadPool.java

```diff
@@ -86,6 +72,84 @@
         public static final String FETCH_SHARD_STORE = "fetch_shard_store";
     }
 
+    public enum ThreadPoolType {
```

Comment:
Can you add javadocs for this, I think a single line for each type that gives a nice description for anyone checking out the code (support peeps) that might not be familiar with java threadpool types would be super helpful